src/pages/index.js

```diff
@@ -163,27 +181,82 @@ profileEditButton.addEventListener("click", () => {
 });
 
 cardModalbtn.addEventListener("click", () => {
-  editModalNameInput.value = profileName.textContent;
-  editModalDescriptionInput.value = profileDescription.textContent;
   openModal(cardModal);
 });
 
+avatarEditButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarForm.addEventListener("submit", handleAvatarSubmit);
 profileForm.addEventListener("submit", handleEditForSubmit);
 cardForm.addEventListener("submit", handleAddCardSubmit);
-
-initialCards.forEach((card) => {
-  const cardElement = getCardElement(card);
-  cardsList.prepend(cardElement);
-});
+deleteForm.addEventListener("submit", handleDeleteSubmit);
+cancelButton.addEventListener("click", () => closeModal(deleteModal));
 
 enableValidation(settings);
 
+function handleEditForSubmit(e) {
+  function makeRequest() {
+    return api
+      .editUserInfo({
+        name: editModalNameInput.value,
+        about: editModalDescriptionInput.value,
+      })
+      .then((userData) => {
+        console.log(userData);
+        profileName.textContent = userData.name;
+        profileDescription.textContent = userData.about;
+        closeModal(editModal);
+      });
+  }
 
-console.log("Hello, world!");
+  handleSubmit(makeRequest, e);
+}
 
-const numbers = [2, 3, 5];
+function handleAddCardSubmit(e) {
+  function makeRequest() {
+    const inputValues = {
+      name: cardNameInput.value,
+      link: cardLinkInput.value,
+    };
+    return api.addNewCard(inputValues).then((card) => {
+      const cardElement = getCardElement(card);
+      cardsList.prepend(cardElement);
+    });
+  }
+
+  handleSubmit(makeRequest, e).finally(() => {
+    closeModal(cardModal);
+    disableButton(cardSubmitBtn, settings);
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**